contracts/smart-account/modules/BatchedSessionRouterModule.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {BaseAuthorizationModule, UserOperation, ISignatureValidator} from "./BaseAuthorizationModule.sol";
+import {ISessionValidationModule} from "./SessionValidationModules/ISessionValidationModule.sol";
+import {ISessionKeyManager} from "../interfaces/ISessionKeyManager.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@account-abstraction/contracts/core/Helpers.sol";
+
+/**
+ * @title Batched Session Router
+ * @dev Built to process executeBatch and executeBatch_y6U calls
+ *         - Every call inside batch should be covered by an appropriate Session Validation Module
+ *         - Parses data provided and sequentially 
+                a) verifies the session key was enabled via SessionKeyManager
+                b) verifies the session key permissions via Session Validation Modules
+ *         - Should be used with carefully verified and audited Session Validation Modules only
+ *         - Compatible with Biconomy Modular Interface v 0.1
+ * @author Fil Makarov - <filipp.makarov@biconomy.io>
+ */
+
+contract BatchedSessionRouter is BaseAuthorizationModule {
+    bytes4 public constant EXECUTE_BATCH_SELECTOR = 0x47e1da2a;
+    bytes4 public constant EXECUTE_BATCH_OPTIMIZED_SELECTOR = 0x00004680;
+
+    /**
+     * @dev validates userOperation. Expects it to be a executeBatch or executeBatch_y6U call
```

Comment:
To be more precise we can use

`Expects userOp.callData to be an executeBatch or executeBatch_y6U call.`

Reply:
done